qiskit_ibm_runtime/ibm_runtime_service.py

```diff
@@ -908,6 +909,31 @@ def program(self, program_id: str, refresh: bool = False) -> RuntimeProgram:
 
         return self._programs[program_id]
 
+    def _retrieve_programs(self, name: str = "") -> None:
+        """Make an API call to fetch programs.
+
+        Args:
+            name: Name of program.
+        """
+        self._programs = {}
```

Comment:
This function should not modify self._programs right? If we call this function with name parameter then we will lose all programs already in cache that don’t match the name. This function should return programs and they should be stored in self._programs in the calling function above when called without name and stored in temp variable when called with name.